### PR TITLE
LPS-194817 [CP 2.0] [BUG] Object "High Priority Contacts" It is populating all projects without a filter

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS-194817](https://liferay.atlassian.net/browse/LPS-194817)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:relevant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 1]
```diff
@@ -49,7 +51,13 @@ const SetupHighPriorityContact = ({
...
 				role: getContactRoleByFilter(filter),
 			},
-			filterRequest: SearchBuilder.eq('contactsCategory', _filter),
+			filterRequest: `${SearchBuilder.eq(
+				'contactsCategory',
+				_filter
+			)} and ${SearchBuilder.eq(
+				'r_accountEntryToHighPriorityContacts_accountEntryERC',
+				project.accountKey
+			)}`,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:60]: It's possible to to that with SeachBuilder constructor

```suggestion
			filterRequest:  new SearchBuilder()
			    .eq('contactsCategory', _filter)
			    .and()
			    .eq('r_accountEntryToHighPriorityContacts_accountEntryERC', project.accountKey)
			    .build(),
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 2]
```diff
@@ -49,7 +51,13 @@ const SetupHighPriorityContact = ({
 				name: `${filter}`,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:51]:
```suggestion
				name: filter,
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ const fetchHeadless = async ({resolveAsJson = true, url}) => {
 const getHighPriorityContacts = async (filter) => {
 	// eslint-disable-next-line @liferay/portal/no-global-fetch
 	const response = await fetch(
-		`${HEADLESS_BASE_URL}${`c/highprioritycontactses/?nestedFields=user&filter=${filter}`}`,
+		`${HEADLESS_BASE_URL}${`c/highprioritycontactses/?nestedFields=userandaccountKey&filter=${filter}`}`,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js:31]: is this right? userandaccountKey?

```suggestion
		`${HEADLESS_BASE_URL}${`c/highprioritycontactses/?nestedFields=userandaccountKey&filter=${filter}`}`,
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx [hunk 4]
```diff
@@ -56,10 +73,13 @@ const IncidentContactCard = ({
...
-				/\s/g,
-				''
-			)}'`,
+			filterRequest: `${SearchBuilder.eq(
+				'contactsCategory',
+				lowerCaseFirstLetter.replace(/\s/g, '')
+			)} and ${SearchBuilder.eq(
+				'r_accountEntryToHighPriorityContacts_accountEntryERC',
+				project.accountKey
+			)}`,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx:82]: Use new SearchBuildeR() constructor as suggested before

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx [hunk 5]
```diff
@@ -245,15 +265,15 @@ const IncidentContactCard = ({
 									>
 										{hasCriticalIncidentContact ? (
 											criticalIncidentContacts
-										) : (
+										) : hasAdministratorRole ? (
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx:268]:
```suggestion
										) : hasAdministratorRole && (
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx [hunk 6]
```diff
@@ -245,15 +265,15 @@ const IncidentContactCard = ({
...
 											<IncidentContactsButton
 												onClick={() =>
 													handleOnClick(
 														HIGH_PRIORITY_CONTACT_CATEGORIES.criticalIncident
 													)
 												}
 											/>
-										)}
+										) : null}
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx:276]:
```suggestion
										)}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx [hunk 7]
```diff
@@ -329,15 +351,15 @@ const IncidentContactCard = ({
 											>
 												{hasPrivacyBreachContact ? (
 													privacyBreachContacts
-												) : (
+												) : hasAdministratorRole ? (
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx:354]:
```suggestion
												) : hasAdministratorRole && (
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx [hunk 8]
```diff
@@ -329,15 +351,15 @@ const IncidentContactCard = ({
...
 													<IncidentContactsButton
 														onClick={() =>
 															handleOnClick(
 																HIGH_PRIORITY_CONTACT_CATEGORIES.privacyBreach
 															)
 														}
 													/>
-												)}
+												) : null}
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/index.jsx:362]:
```suggestion
												)}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx [hunk 9]
```diff
@@ -118,10 +118,13 @@ const TeamMembersTable = ({
...
 			},
-			filterRequest: SearchBuilder.eq(
+			filterRequest: `${SearchBuilder.eq(
 				'contactsCategory',
 				lowerCaseFirstLetter.replace(/\s/g, '')
-			),
+			)} and ${SearchBuilder.eq(
+				'r_accountEntryToHighPriorityContacts_accountEntryERC',
+				koroneikiAccount.accountKey
+			)}`,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx:127]: use new SearchBuilder constructor

**Review by kevenleone — CHANGES_REQUESTED** [review]
Good job on that @eduardo-diniz , can you see my suggestions?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx (hunk 1 above)
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:60]: Nice!

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by kevenleone** [comment]
ci:forward

**Comment by eduardo-diniz** [comment]
ci:test:relevant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx [hunk 10]
```diff
@@ -118,10 +118,13 @@ const TeamMembersTable = ({
...
-				'contactsCategory',
-				lowerCaseFirstLetter.replace(/\s/g, '')
-			),
+			filterRequest: new SearchBuilder()
+				.eq('contactsCategory', lowerCaseFirstLetter.replace(/\s/g, ''))
+				.and()
+				.eq(
+					'r_accountEntryToHighPriorityContacts_accountEntryERC',
+					koroneikiAccount.accountKey
+				),
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx:127]: build() is missing

```suggestion
			filterRequest: new SearchBuilder()
				.eq('contactsCategory', lowerCaseFirstLetter.replace(/\s/g, ''))
				.and()
				.eq(
					'r_accountEntryToHighPriorityContacts_accountEntryERC',
					koroneikiAccount.accountKey
				)
				.build(),
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Review by kevenleone — APPROVED** [review]

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by kevenleone** [comment]
resent: https://github.com/brianchandotcom/liferay-portal/pull/140642#issuecomment-1717943788

**Pull request closed without merging**